lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1625,6 +1636,220 @@ func parseAddrType(addrType AddressType,
 	}
 }
 
+// Text used to signify that a signed message follows and to prevent
+// inadvertently signing a transaction or a signature.
+// Same as bitcoin core and btcd.
+const messageSignatureHeader = "Bitcoin Signed Message:\n"
+
+// SignMessageWithAddr signs a message with the private key of the provided
+// address. The address needs to belong to the lnd wallet.
+func (w *WalletKit) SignMessageWithAddr(ctx context.Context,
+	req *SignMessageWithAddrRequest) (
+	*SignMessageWithAddrResponse, error) {
+
+	addr, err := btcutil.DecodeAddress(req.Addr, w.cfg.ChainParams)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode address %v", err)
+	}
+
+	if !addr.IsForNet(w.cfg.ChainParams) {
+		return nil, fmt.Errorf("encoded address is for"+
+			"the wrong network %v", addr.EncodeAddress())
+	}
+
+	// Fetch Address Infos from own wallet and check whether it belongs
+	// to the lnd wallet
+	managedAddr, err := w.cfg.Wallet.AddressInfo(addr)
+	if err != nil {
+		return nil, fmt.Errorf("address could not be found in the "+
+			"wallet database %w", err)
+	}
+
+	// Verify that we own the private key to the provided address
+	pubKey, ok := managedAddr.(waddrmgr.ManagedPubKeyAddress)
+	if !ok {
+		return nil, fmt.Errorf("private key to address is unknown")
+	}
+
+	// TODO: Cannot use the signer here because its restricted to the 1017
+	// scope Make the change in keychain to allow the signer to depend
+	// on scope and purpose?
+
+	// To prevent attack vectors e.g. signing an already signed signature
+	// a prefix is attached (as bitcoin-core and btcd does).
+	var buf bytes.Buffer
+	err = wire.WriteVarString(&buf, 0, messageSignatureHeader)
```

Comment:
`WriteVarString` writes the length of the signature header first and then the signature header. I think you only want to use the header bytes here and not prepend the length. You could use `buf.WriteString`.

Reply:
Staying bitcoin-core/btcd compliant we need to use WriteVarString here. See comment below.

Reply:
OK